src/components/PageButtons.vue

```diff
@@ -0,0 +1,71 @@
+<script setup>
+import IconHome from './icons/IconHome.vue';
+
+const props = defineProps({ buttons: { type: Array, required: true } })
+
+
+</script>
+
+<template>
+  <div class="container">
+    <button
+      v-for="(button, index) in buttons"
+      :key="index"
+      class="page"
+    >
+      <aside class="icon-wrapper">
+        <component :is="button.comp" />
+      </aside>
+
+      <span class="page-text">
+        {{ button.name }}
+      </span>
+    </button>
+  </div>
+</template>
```

Comment:
Ok, so now that you've made a component that handles _all_ of the buttons, we now need to make components for a single button.